myia/opt/lib.py

```diff
@@ -819,6 +820,28 @@ def caller_is_marked(g, node, args):
 )
 
 
+@pattern_replacer("just", X, interest=None)
+def expand_composite(resources, node, equiv):
+    """Expand a composite primitive."""
+    if node.is_apply():
+        operation = node.inputs[0]
+        parameters = node.inputs[1:]
+        if operation.is_constant() and isinstance(
+            operation.value, CompositePrimitive
+        ):
+            from myia.parser import parse
+            from myia.ir.anf import Graph
+
+            py_impl = operation.value.defaults()["python_implementation"]
+            g = parse(py_impl)
```

Comment:
This is potentially wasteful: if the composite is used multiple times, we don't want to parse it repeatedly. In any case, there's already a way to incorporate new graphs during the optimization phase, which is used in `expand_J`. More precisely [this line](https://github.com/mila-iqia/myia/blob/60a4bd8f7b0bcb08fb730db52d2dd316e4d94f9b/myia/opt/lib.py#L1148).